### PR TITLE
exclude go modules for dependbot

### LINKED ISSUE #1133: Exclude the go models for dependabot
/kind cleanup
/area provider/ibmcloud

**What steps did you take and what happened:**
We have seen that dependabot goes and update all the go module in the repository but we don't expect dependabot to update certain repos automatically, e.g: k8s, ginkgo modules etc.. lets exclude these set of modules to avoid generating noise.

**What did you expect to happen:**

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api version:
- Minikube/KIND version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by Prajyot-Parab**
**What this PR does / why we need it**:
- exclude go modules for dependbot

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1133 

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
exclude go modules for dependbot
```

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -4,14 +4,32 @@ updates:
...
   # Enable version updates for Go tools
   - package-ecosystem: "gomod"
     directory: "/hack/tools"
     schedule:
       interval: "weekly"
-
-  - package-ecosystem: "docker"
-    directory: "/"
-    schedule:
-      interval: "weekly"
```
Comment (mkumatag) [inline .github/dependabot.yml:17]: why is this removed?

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: .github/dependabot.yml (hunk 1 above)
Comment (Prajyot-Parab) [reply .github/dependabot.yml:17]: This takes care of updating only `golang` in Dockerfile which anyways we update manually, so seems like irrelevant.

Comment (mkumatag) [reply .github/dependabot.yml:17]: yes, that's the intention of this code to update the golang version in the dockerfile but seems like it is not working as expected and the reason why we are doing it manually. Lets keep this line as is for now and explore later why it is not updating the version(of course only the patched version)

Comment (Prajyot-Parab) [reply .github/dependabot.yml]: Done.

**Review by mkumatag — APPROVED** [review]
/lgtm
